astropy/cosmology/core.py

```diff
@@ -135,11 +131,13 @@ def __init__(self, *args, name=None, meta=None, **kwargs):
         self._name = name
         self.meta.update(meta or {})
 
-    # Make initial signature of cosmology. Overwritten in subclasses.
-    # (Need to make one here b/c Cosmology is not abstract).
-    _init_signature = signature(__init__)
-    _init_signature = _init_signature.replace(
-        parameters=list(_init_signature.parameters.values())[1:])
+    @classproperty(lazy=True)
+    def _init_signature(cls):
+        """Initial signature of Cosmology. Overwritten in subclasses."""
```

Comment:
I think the docstring is not quite right any more. Just "Initialization signature of the class". Or maybe just omit?

Reply:
👍